kusto/kusto.go

```diff
@@ -28,80 +24,16 @@ type queryer interface {
 // Authorization provides the ADAL authorizer needed to access the resource. You can set Authorizer or
 // Config, but not both.
 type Authorization struct {
-	// Authorizer provides an authorizer to use when talking to Kusto. If this is set, the
-	// Authorizer must have its Resource (also called Resource ID) set to the endpoint passed
-	// to the New() constructor. This will be something like "https://somename.westus.kusto.windows.net".
-	// This package will try to set that automatically for you.
-	Authorizer autorest.Authorizer
-	// Config provides the authorizer's config that can create the authorizer. We recommending setting
-	// this instead of Authorizer, as we will automatically set the Resource ID with the endpoint passed.
-	Config auth.AuthorizerConfig
+	// Token provider that can be used to get the access token.
+	tokenProvider tokenProvider
 }
 
 // Validate validates the Authorization object against the endpoint an preps it for use.
 // For internal use only.
-func (a *Authorization) Validate(endpoint string) error {
+func (a *Authorization) Validate() error {
```

Comment:
We support no-auth mode for some cases (dev mostly).
Not sure if other SDKs enforce this.

@AsafMah  WDYT?

Reply:
All other sdks have no auth mode, we should have it here too.